amqp/src/test/scala/akka/stream/alpakka/amqp/scaladsl/AmqpConnectionProvidersSpec.scala

```diff
@@ -135,9 +147,57 @@ class AmqpConnectionProvidersSpec extends AmqpSpec {
       connection1.isOpen should be(false)
       connection2.isOpen should be(false)
     }
+
+    "registers and unregisters a single connection shutdown hook per graph" in {
+      val spyConnectionPromise = Promise[Connection]
+      val shutdownsAdded = new AtomicInteger()
+      val shutdownsRemoved = new AtomicInteger()
+
+      val connectionFactory = new ConnectionFactory()
+      val connectionProvider = DelegatingConnectionProvider(
+        AmqpConnectionFactoryConnectionProvider(connectionFactory, List(("localhost", 5672))),
+        conn => {
+          val proxy = new AmqpProxyConnection(conn) {
+            override def addShutdownListener(shutdownListener: ShutdownListener): Unit = {
+              shutdownsAdded.incrementAndGet()
+              super.addShutdownListener(shutdownListener)
+            }
+
+            override def removeShutdownListener(shutdownListener: ShutdownListener): Unit = {
+              shutdownsRemoved.incrementAndGet()
+              super.removeShutdownListener(shutdownListener)
+            }
+          }
+
+          spyConnectionPromise.success(proxy)
+          proxy
+        }
+      )
+      
+      val reusableConnectionProvider =
+        AmqpCachedConnectionProvider(provider = connectionProvider, automaticRelease = true)
+
+      def queueName = "amqp-conn-prov-it-spec-simple-queue-" + System.currentTimeMillis()
+      val queueDeclaration = QueueDeclaration(queueName)
+
+      val amqpSink = AmqpSink.simple(
+        AmqpSinkSettings(reusableConnectionProvider)
+          .withRoutingKey(queueName)
+          .withDeclarations(queueDeclaration)
+      )
+
+      val input = Vector("one", "two", "three", "four")
+
+      Future.traverse(input)(in => Source.single(ByteString(in)).runWith(amqpSink))
```

Comment:
Given my [earlier issues](https://github.com/akka/alpakka/pull/887#issuecomment-380319279) I'm wondering if this isn't a valid test since I get inconsistent results with `autoRelease = false`